### PR TITLE
593 natural ventilation toggle in hizard

### LINKED ISSUE #593: Building Input Form - Toggle natural ventilation
Students can see the impact of using an albeit simplified implementation of natural ventilation on demand.

Would involve some WinForms (checkbox in ventilation tab), changing some input params for sia380 maybe?

### PR DESCRIPTION
**Pull request by christophwaibel**
# Natural ventilation toggle in hizard

## Issues

Closes #593

## Description

- adds a new checkbox in the Ventilations tab of the Building hizard. This passes on a boolean to either switch on or off natural ventilation in the sia380 component within the core. Similar to the Adaptive Comfort checkbox

## Checklist

- [ ] Remove debugging artifacts (if needed)
- [ ] Rebuild the Setup_Hive.exe
- [ ] Update Hive Wiki (if needed)
- [ ] Update most important Grasshopper template(s); as of now [/GrasshopperExamples/LectureExercises/EaCS3_E04_Hive_Template.gh](https://github.com/architecture-building-systems/hive/blob/master/GrasshopperExamples/LectureExercises/EaCS3_E04_Hive_Template.gh)

### CODE REVIEW

**Comment by christophwaibel** [comment]
What works:

image

- the checkbox for `Natural Ventilation?` and `Adaptive?`, correctly passing on booleans into the core.

What does not work:

- When saving the `.gh` file, the checkbox values are not remembered when re-opening the file. This should work similarly to how changed U-values, or selected building type etc. are saved in the hizard.
- In the SIA main tab of the building hizard, `Heating`, `Cooling`, `Heat Setback`, `Cool Setback` values are not saved, not even within one grasshopper session (which works for the two checkboxes). I.e., changing a setback value is not stored after closing the hizard.

**Comment by christophwaibel** [comment]
@daren-thomas any pointers on what we might be missing? <3

File: setup/build.cmd [hunk 1]
```diff
@@ -4,11 +4,11 @@
 @REM Requires python 2.7, likely from conda env.
 SET PYTHON=python
 SET IPY=%PROGRAMFILES%\IronPython 2.7\ipy.exe
-SET HB=C:\Users\Christoph\Documents\GitHub\honey-badger\honey-badger.py
+SET HB=C:\Users\chwaibel\Documents\GitHub\honey-badger\honey-badger.py
 SET MSBUILD=%PROGRAMFILES(x86)%\Microsoft Visual Studio\2019\Community\MSBuild\Current\Bin\MSBuild.exe
 SET MAKENSIS=%PROGRAMFILES(X86)%\NSIS\makensis.exe
 
-SET HIVEWIKIPY=C:\Users\Christoph\Documents\GitHub\hive.wiki\hbwiki.py
+SET HIVEWIKIPY=C:\Users\chwaibel\Documents\GitHub\hive.wiki\hbwiki.py
```
Comment (daren-thomas) [inline setup/build.cmd:11]: you can probably use environment variables here: `%USERPROFILE%` refers to your home directory. BUT: see [here on stackoverflow](https://stackoverflow.com/questions/17063947/get-current-batchfile-directory) for a more interesting idea: `%~dp0` refers to the drive and the directory of the batch file itself - this is useful, when you're not the only developer...

**Review by daren-thomas — COMMENTED** [review]
hmmm... nope. no real idea. you'll need to step through to see if the changes are actually being loaded. where are you expecting to save them to? has to be a GH_Component - maybe debug that component with breakpoints on the `Write` and `Read` methods. Do they load the proper data?

File: setup/build.cmd (hunk 1 above)
Comment (maxenceryan) [reply setup/build.cmd:11]: ah cool thanks for the tip!

**Comment by maxenceryan** [comment]
so i managed to correct the heating setpoints / setbacks such that they persist in session and when closing / reopening. I forgot some event handling via this Validating thing in Winforms, along with some brush and fontweight fields.

With the checkboxes, couldn't fix it but it seems:
- we attempt to store it both in zone and siaRoom, which brings up the conflation of these two classes that we have so far. i think original intent was to use the siaRoom / record as a static lookup but it has other stuff in there that are hive based now (such as running adaptive comfort or not) which i think is confusing the purpose of that class. i dunno the best approach, but maybe some separation of concerns needed here? it is also stored in zone though
- when reading/writing files, only the siaRoom seems to be serialized and deserialized. This means that the zone is recreated with whatever is inputted into the hizard upon loading the file. So properties like Zone.RunAdaptiveComfort will intiialise with a default in the code (not what was saved before), as we only feed in geometry to initialise Zones (from what I gather). So the issue is those properties we had saved in a previous session in the hizard, and then written to the zone, would not persist to another session. 
- But since those properties are also (in parallel) saved to SiaRoom.RunAdaptiveComfort, those should persist but for some reason dont. It seems upon writing the grasshopper file, there is a mismatch between RunAdaptiveComfort and RunNaturalVentialation in the Zone and SiaRoom classes. Zone props are updated via hizard but SiaRoom props are not. Could be that the lookup is overriding what is assigned in the hizard, but i dont know.

**Comment by christophwaibel** [comment]
let's review this and keep the non-persisting checkboxes for a future issue

**Comment by maxenceryan** [comment]
Issue created for persisting checkboxes #644

**Review by maxenceryan — APPROVED** [review]

**Pull request merged**